src/object/srv_obj.c

```diff
@@ -975,6 +975,14 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 	if (obj_rpc_is_update(rpc)) {
 		obj_singv_ec_rw_filter(orw, iods, offs, true);
 		bulk_op = CRT_BULK_GET;
+
+		/** Fault injection - corrupt data from network */
+		if (DAOS_FAIL_CHECK(DAOS_CSUM_CORRUPT_UPDATE)  && !rma) {
```

Comment:
it is only to corrupt data for !rma (RPC inline)? actually bulk data also can be corrupt if you want, crt_bulk_access() can get the sgl and then corrupt it.

Reply:
True. Bulk data will get corrupted in obj_bulk_transfer